common/types.go

```diff
@@ -55,6 +55,24 @@ func (h Hash) Bytes() []byte { return h[:] }
 func (h Hash) Big() *big.Int { return new(big.Int).SetBytes(h[:]) }
 func (h Hash) Hex() string   { return hexutil.Encode(h[:]) }
 
+// TerminalString implements log.TerminalStringer, formatting a string for console
+// output during logging.
+func (h Hash) TerminalString() string {
+	return fmt.Sprintf("%x…%x", h[:2], h[30:])
+}
```

Comment:
Please add more than 2 bytes of context at the beginning of the hash. IMHO the beginning is more important than the end.